### PR TITLE
feat(crons): Add active incident index to MonitorIncident model

### PR DESCRIPTION
**Pull request by davidenwang**
To allow us to quickly query for active incidents

https://github.com/getsentry/sentry/pull/65963#discussion_r1513633297

### CODE REVIEW

File: migrations_lockfile.txt [hunk 1]
```diff
@@ -9,5 +9,5 @@ feedback: 0004_index_together
 hybridcloud: 0013_add_orgauthtokenreplica_token_index
 nodestore: 0002_nodestore_no_dictfield
 replays: 0004_index_together
-sentry: 0664_create_new_broken_monitor_detection_table
+sentry: 0665_add_active_monitor_incident_index
```
Comment (evanpurkhiser) [inline migrations_lockfile.txt:12]: Will need a rebase after https://github.com/getsentry/sentry/pull/66429

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/migrations/0665_add_active_monitor_incident_index.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
```
Comment (wedamija) [inline src/sentry/migrations/0665_add_active_monitor_incident_index.py:19]: This table is large enough that adding this index will probably time out during deploy. Let's set this to dangerous. You can then run it yourself: https://www.notion.so/sentry/Running-Dangerous-Migrations-859a553040fc4649b1f8dc27055c8619?pvs=4#39e52a7d48a64895bf9287dc1f4af581

Comment (davidenwang) [reply src/sentry/migrations/0665_add_active_monitor_incident_index.py:19]: dang good call, 3M+ rows on this table 0.0

**Review by wedamija — APPROVED** [review]
lgtm. You can merge this, but we'll need to hold off actually running it until the db switch is complete

**Pull request merged**